scipy/optimize/tests/test_bracket.py

```diff
@@ -345,6 +345,40 @@ def f(x):
                                 xmin=1)
         assert not res.success
 
+        # 5. Multiple searches in same direction terminate
+        # simultaneously in some iteration after the corresponding searches
+        # in the other direction terminated without finding a root.
+
+        # Example based on that in
+        # https://github.com/scipy/scipy/pull/22560#discussion_r1962853839
+        def f(x, p):
+            return np.exp(x) - p
+
+        p = np.asarray([0.29, 0.35])
+        res = _bracket_root(f, xl0=-1, xmin=-np.inf, xmax=0, args=(p, ))
+
+        # 6. https://github.com/scipy/scipy/pull/22560/files#r1962952517
+        def f(x, p, c):
+            return np.exp(x*c) - p
+
+        p = [0.32061201, 0.39175242, 0.40047535, 0.50527218, 0.55654373,
```

Comment:
subnormal nit: I think this will still fail (but can look a little more compact) if we chop some digits off.

Reply:
True. I'm off for the rest of the day but could do it tomorrow if you think it's important enough.